helix-core/src/main/java/org/apache/helix/controller/stages/MessageGenerationPhase.java

```diff
@@ -445,7 +448,8 @@ private Message createStateTransitionCancellationMessage(HelixManager manager, R
       message.setFromState(fromState);
       message.setToState(toState);
       message.setTgtSessionId(sessionId);
-      message.setSrcSessionId(manager.getSessionId());
+      message.setSrcSessionId(managerSessionId);
+      message.setExpectedSessionId(managerSessionId);
```

Comment:
What is the difference between SrcSessionId and ExpectedSessionId?

Reply:
The difference I suppose is: say, a message is forwarded/sent from node1 to node2. So srcSessionId = sessionNode1. This message doesn't have to be written by node1's zkclient, but another zkclient `session-0`. So expectedSessionId = `session-0`. I'd like to differentiate  srcSessionId and write expectedSessionId.
Let me know I miss anything.